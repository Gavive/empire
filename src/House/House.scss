$color-primary: #224889;
$color-roof: #A6CFFF;
$color-ledge: #79AAFF;
$color-window-left: #65EBFF;
$color-window-right: #71FBFF;
$color-window: linear-gradient(to right, $color-window-left, $color-window-left 49.9%, $color-window-right 50%, $color-window-right);
$color-shadow: #E1EAFF;
$duration: 500ms;
$delay: 70ms;
$easing: cubic-bezier(.1, 0, .3, 1);

$parts: (
  6: (
    wings: 425px,
    front: 150px,
  ),
  5: (
    wings: 355px,
    front: 150px,
  ),
  4: (
    wings: 300px,
    front: 125px,
  ),
  3: (
    wings: 240px,
    front: 150px,
  ),
);

.house {
  height: 225px;
  width: 520px;
}