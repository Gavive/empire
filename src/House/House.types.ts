export type Props = {};

export type State = {
  roomCount: number;
};
