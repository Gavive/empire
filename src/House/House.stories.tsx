import React from "react";
import House from "./House";
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';

storiesOf("House", module)
  .addDecorator(centered)
  .add("Default", () => <House />)
