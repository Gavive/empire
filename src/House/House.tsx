import React, { Component } from "react";
import "./House.scss";

export default class House extends Component {

  public render(): React.ReactNode {
    return (
      <div className="house">
      </div>
    );
  }
}
