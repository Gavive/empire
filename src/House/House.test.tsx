import React from "react";
import { render, screen } from "@testing-library/react";

import House from "./House";
import { Props } from "./House.types";

describe("🏠 House", () => {
  let props: Props;

  beforeEach(() => {
    props = {
      theme: "primary"
    };
  });

  const renderComponent = () => render(<House {...props} />);

  it("to match snapshot", () => {
    const component = renderComponent();

    expect(component).toMatchSnapshot();
  })

  it("container has the correct elements", () => {
    const component = renderComponent();
    const container = component.container;
    const houseDiv = component.container.getElementsByTagName("div").item(0);
    const houseLabel = component.container.getElementsByTagName("label").item(0);
    const houseInput = component.container.getElementsByTagName("input").item(0);

    expect(container).toBeInTheDocument();
    expect(container.hasChildNodes()).toBe(true);
    expect(container.childElementCount).toBe(3);

    expect(houseDiv).toHaveClass("house");
    expect(houseDiv?.id).toEqual("house");
    expect(houseDiv?.getAttribute("data-rooms")).toEqual("1");

    expect(houseLabel).toHaveClass("house-label");
    expect(houseLabel?.id).toEqual("label");
    expect(houseLabel?.getAttribute("for")).toEqual("range");
    expect(houseLabel?.getAttribute("data-rooms")).toEqual("1");

    expect(houseInput).not.toHaveClass();
    expect(houseInput?.id).toBe("range");
    expect(houseInput?.getAttribute("min")).toBe("1");
    expect(houseInput?.getAttribute("max")).toBe("6");
    expect(houseInput?.getAttribute("step")).toBe("1");
    expect(houseInput?.getAttribute("value")).toBe("1");
    expect(houseInput?.getAttribute("type")).toBe("range");
    expect(houseInput?.getAttribute("onChange")).toBeDefined();
  });
});
