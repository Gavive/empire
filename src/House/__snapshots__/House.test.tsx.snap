// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`üè† House to match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="house"
        data-rooms="1"
        id="house"
      >
        <div
          class="house-wings"
          data-flip-key="wings"
        >
          <div
            class="house-left-wing"
          >
            <div
              class="house-window"
            />
            <div
              class="house-window"
            />
            <div
              class="house-sparkle"
            >
              <div
                class="house-sparkle-dots"
              />
            </div>
          </div>
          <div
            class="house-right-wing"
          >
            <div
              class="house-window"
            />
            <div
              class="house-window"
            />
            <div
              class="house-sparkle"
            >
              <div
                class="house-sparkle-dots"
              />
            </div>
          </div>
          <div
            class="house-roof"
          >
            <div
              class="house-ledge"
            />
          </div>
        </div>
        <div
          class="house-front"
          data-flip-key="front"
        >
          <div
            class="house-chimney"
          />
          <div
            class="house-facade"
          />
          <div
            class="house-window"
          >
            <div
              class="house-sparkle"
            >
              <div
                class="house-sparkle-dots"
              />
            </div>
          </div>
          <div
            class="house-doorway"
          >
            <div
              class="house-stairs"
            />
            <div
              class="house-door"
            />
          </div>
          <div
            class="house-gable"
          >
            <div
              class="house-roof"
            >
              <div
                class="house-ledge"
              />
            </div>
          </div>
        </div>
      </div>
      <label
        class="house-label"
        data-rooms="1"
        for="range"
        id="label"
      >
        Rooms
      </label>
      <input
        id="range"
        max="6"
        min="1"
        step="1"
        type="range"
        value="1"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="house"
      data-rooms="1"
      id="house"
    >
      <div
        class="house-wings"
        data-flip-key="wings"
      >
        <div
          class="house-left-wing"
        >
          <div
            class="house-window"
          />
          <div
            class="house-window"
          />
          <div
            class="house-sparkle"
          >
            <div
              class="house-sparkle-dots"
            />
          </div>
        </div>
        <div
          class="house-right-wing"
        >
          <div
            class="house-window"
          />
          <div
            class="house-window"
          />
          <div
            class="house-sparkle"
          >
            <div
              class="house-sparkle-dots"
            />
          </div>
        </div>
        <div
          class="house-roof"
        >
          <div
            class="house-ledge"
          />
        </div>
      </div>
      <div
        class="house-front"
        data-flip-key="front"
      >
        <div
          class="house-chimney"
        />
        <div
          class="house-facade"
        />
        <div
          class="house-window"
        >
          <div
            class="house-sparkle"
          >
            <div
              class="house-sparkle-dots"
            />
          </div>
        </div>
        <div
          class="house-doorway"
        >
          <div
            class="house-stairs"
          />
          <div
            class="house-door"
          />
        </div>
        <div
          class="house-gable"
        >
          <div
            class="house-roof"
          >
            <div
              class="house-ledge"
            />
          </div>
        </div>
      </div>
    </div>
    <label
      class="house-label"
      data-rooms="1"
      for="range"
      id="label"
    >
      Rooms
    </label>
    <input
      id="range"
      max="6"
      min="1"
      step="1"
      type="range"
      value="1"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
